src/middlewared/middlewared/plugins/network.py

```diff
@@ -641,8 +641,7 @@ async def __restore_datastores(self):
         # breaks `failover.status` on TrueNAS HA systems. This means the
         # following `datastore.insert` operations fail because we can't
         # determine the Active/Standby controller.
-        if await self.middleware.call('failover.licensed'):
-            await self.middleware.call('failover.sync_to_peer')
+        await self.middleware.call_hook('rollback.post_sync')
```

Comment:
Here is an idea, how about we put it within the rollback method, after __restore_datastores call and name it `interface.pre_rollback` so the hook name is consistent with the others? We try to name it similar to the plugin/service

Reply:
I agree, but you mean `interface.post_rollback` right since we're running after the rollback operation occurs?

Reply:
Oh I see, you want it to run after the `__restore_datastores` method but before the `sync` method?

Reply:
Its a grey line, maybe.. we do rollback the internal state, but dont actually rollback until after .sync runs
I am fine with either